### PR TITLE
add snapcraft 24.04 aliases

### PR DESCRIPTION
**Pull request by andrei-toterman**

### CODE REVIEW

File: src/platform/platform_shared.h [hunk 1]
```diff
@@ -23,8 +23,15 @@
...
-const std::unordered_set<std::string>
-    supported_snapcraft_aliases{"core18", "18.04", "core20", "20.04", "core22", "22.04", "devel"};
+const std::unordered_set<std::string> supported_snapcraft_aliases{"core18",
+                                                                  "18.04",
+                                                                  "core20",
+                                                                  "20.04",
+                                                                  "core22",
+                                                                  "22.04",
+                                                                  "core24",
+                                                                  "24.04"
```
Comment (ricab) [inline src/platform/platform_shared.h:33]: missing a comma ;)

**Review by ricab — CHANGES_REQUESTED** [review]

File: src/platform/platform_shared.h (hunk 1 above)
Comment (andrei-toterman) [reply src/platform/platform_shared.h:33]: oopsie

**Review by ricab — COMMENTED** [review]
Looks good, thanks. I'll just want to give it a spin on a couple of platforms before merging.

**Review by ricab — APPROVED** [review]
Worked smoothly, thanks again.

**Pull request merged**